src/main/java/nl/novi/techiteasy/services/TelevisionService.java

```diff
@@ -48,10 +62,52 @@ public TelevisionDto updateTelevision(Long id, TelevisionInputDto television) {
         return televisionMapper.toTelevisionDto(televisionRepository.save(updatedTelevision));
     }
 
-
     public TelevisionDto deleteTelevision(Long id) {
         televisionRepository.deleteById(id);
         return null;
     }
 
+    public TelevisionDto assignRemoteControllerToTelevision(long id, IdInputDto remoteControllerId) {
+        Long getRemoteControllerId = remoteControllerId.getId();
+        Optional<RemoteController> remoteControllerOptional = remoteControllerRepository.findById(getRemoteControllerId);
+        Optional<Television> televisionOptional = televisionRepository.findById(id);
+        if (remoteControllerOptional.isPresent() && televisionOptional.isPresent()) {
+            RemoteController remoteController = remoteControllerOptional.get();
+            Television television = televisionOptional.get();
+            television.setRemoteController(remoteController);
+            return televisionMapper.toTelevisionDto(televisionRepository.save(television));
+        } else {
+            throw new RecordNotFoundException("Remote controller with id " + remoteControllerId + " not found");
+        }
+    }
+
+    public TelevisionDto assignCiModuleToTelevision(long id, IdInputDto ciModuleId) {
+        Long getCiModuleId = ciModuleId.getId();
+        Optional<CiModule> ciModuleOptional = ciModuleRepository.findById(getCiModuleId);
+        Optional<Television> televisionOptional = televisionRepository.findById(id);
+        if (ciModuleOptional.isPresent() && televisionOptional.isPresent()) {
+            CiModule ciModule = ciModuleOptional.get();
+            Television television = televisionOptional.get();
+            television.setCiModule(ciModule);
+            return televisionMapper.toTelevisionDto(televisionRepository.save(television));
+        } else {
+            throw new RecordNotFoundException("Ci module with id " + ciModuleId + " not found");
+        }
+    }
+
+    public TelevisionDto assignWallBracketToTelevision(long id, IdInputDto wallBracketId) {
+        Long getWallBracketId = wallBracketId.getId();
+        List<WallBracket> wallBrackets = new ArrayList<>();
+        Optional<WallBracket> wallBracketOptional = wallBracketRepository.findById(getWallBracketId);
```

Comment:
In de methode assignWallBracketToTelevision gebruik je een nieuwe lijst (List<WallBracket> wallBrackets = new ArrayList<>();) en voeg je daar het opgehaalde WallBracket object aan toe. Vervolgens voeg je deze lijst toe aan de televisie met television.setWallBracket(wallBrackets). Dit vervangt mogelijk elke keer de bestaande lijst met een nieuwe lijst waarin telkens slechts één WallBracket staat.
Misschien kun je de in plaats van elke keer een nieuwe lijst aan te maken, de bestaande lijst van wall brackets van de televisie ophalen en daar de nieuwe wall bracket aan toevoegen.